MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/TurfRingActivity.java

```diff
@@ -0,0 +1,166 @@
+ * (i.e. ring) around a center coordinate.
+ */
+public class TurfRingActivity extends AppCompatActivity implements MapboxMap.OnMapClickListener {
+  private static final String OUTER_CIRCLE_GEOJSON_SOURCE_ID = "OUTER_CIRCLE_GEOJSON_SOURCE_ID";
+  private static final String INNER_CIRCLE_GEOJSON_SOURCE_ID = "INNER_CIRCLE_GEOJSON_SOURCE_ID";
+  private static final String OUTER_CIRCLE_LAYER_ID = "OUTER_CIRCLE_LAYER_ID";
+  private static final String INNER_CIRCLE_LAYER_ID = "INNER_CIRCLE_LAYER_ID";
+  private static final int OUTER_CIRCLE_MILE_RADIUS = 1;
+  private static final double MILE_DIFFERENCE_BETWEEN_CIRCLES = .2;
+  private static final int CIRCLE_STEPS = 360;
+  private static final Point POINT_IN_MIDDLE_OF_CIRCLE = Point.fromLngLat(-115.150738, 36.16218);
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_lab_hollow_circle);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        TurfRingActivity.this.mapboxMap = mapboxMap;
+        mapboxMap.setStyle(new Style.Builder()
+            .fromUrl(Style.LIGHT)
+            .withSource(new GeoJsonSource(OUTER_CIRCLE_GEOJSON_SOURCE_ID))
+            .withSource(new GeoJsonSource(INNER_CIRCLE_GEOJSON_SOURCE_ID))
+            .withLayer(new FillLayer(OUTER_CIRCLE_LAYER_ID, OUTER_CIRCLE_GEOJSON_SOURCE_ID).withProperties(
+              fillColor(Color.RED)
+            ))
+            .withLayer(new FillLayer(INNER_CIRCLE_LAYER_ID, INNER_CIRCLE_GEOJSON_SOURCE_ID).withProperties(
+              fillColor(Color.RED)
+            )), new Style.OnStyleLoaded() {
+              @Override
+              public void onStyleLoaded(@NonNull Style style) {
+                moveRing(POINT_IN_MIDDLE_OF_CIRCLE);
+
+                TurfRingActivity.this.mapboxMap.addOnMapClickListener(TurfRingActivity.this);
+
+                Toast.makeText(TurfRingActivity.this, getString(R.string.tap_on_map),
+                  Toast.LENGTH_SHORT).show();
+              }
+            }
+        );
+      }
+    });
+  }
+
+  @Override
+  public boolean onMapClick(@NonNull LatLng point) {
+
+    moveRing(Point.fromLngLat(point.getLongitude(), point.getLatitude()));
+
+    return true;
+  }
+
+  private void moveRing(Point centerPoint) {
+    if (mapboxMap.getStyle() != null) {
+      Style style = mapboxMap.getStyle();
```

Comment:
just as fyi, alternatively to avoid nullability of a style you can use `mapboxMap.getStyle(OnStyleLoadedCallback)` this will give you a hook where style will always be non null.